opensquirrel/writer.py

```diff
@@ -1,41 +1,37 @@
-from opensquirrel.common import ArgType
-from opensquirrel.gates import querySignature
+from opensquirrel.squirrel_ir import Comment, Float, Gate, Int, Qubit, SquirrelIR, SquirrelIRVisitor, Statement
 
 
-class Writer:
-    NUMBER_OF_SIGNIFICANT_DIGITS = 8
+class _WriterImpl(SquirrelIRVisitor):
+    number_of_significant_digits = 8
 
-    def __init__(self, gates):
-        self.gates = gates
+    def __init__(self, number_of_qubits, qubit_register_name):
+        self.qubit_register_name = qubit_register_name
+        self.output = f"""version 3.0\n\nqubit[{number_of_qubits}] {qubit_register_name}\n\n"""
 
-    @classmethod
-    def _format_arg(cls, squirrelAST, arg, t: ArgType):
-        if t == ArgType.QUBIT:
-            return f"{squirrelAST.qubitRegisterName}[{arg}]"
-        if t == ArgType.INT:
-            return f"{int(arg)}"
-        if t == ArgType.FLOAT:
-            return f"{float(arg):.{Writer.NUMBER_OF_SIGNIFICANT_DIGITS}}"
+    def visit_qubit(self, qubit: Qubit):
+        return f"{self.qubit_register_name}[{qubit.index}]"
 
-        assert False, "Unknown argument type"
+    def visit_int(self, _int: Int):
```

Comment:
For consistency you could use `i` here as well, iso `_int`, as done in the `squirrel_ir.py` module. (same for `_float`).